x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/LimitOperator.java

```diff
@@ -92,8 +93,18 @@ public Page getOutput() {
                 filter[i] = i;
             }
             Block[] blocks = new Block[lastInput.getBlockCount()];
-            for (int b = 0; b < blocks.length; b++) {
-                blocks[b] = lastInput.getBlock(b).filter(filter);
+            boolean success = false;
+            try {
+                for (int b = 0; b < blocks.length; b++) {
+                    blocks[b] = lastInput.getBlock(b).filter(filter);
+                }
+                success = true;
+            } finally {
+                Releasables.closeExpectNoException(lastInput::releaseBlocks);
+                lastInput = null;
+                if (success == false) {
+                    Releasables.closeExpectNoException(blocks);
```

Comment:
Could this (and the similar in `FilterOperator`) not triger a NPE if  `blocks` doesn't get to be fully initialized?